src/Access/HTTPAuthClient.h

```diff
@@ -80,10 +84,21 @@ class HTTPBasicAuthClient : private HTTPAuthClient<TResponseParser>
     using HTTPAuthClient<TResponseParser>::HTTPAuthClient;
     using Result = HTTPAuthClient<TResponseParser>::Result;
 
-    Result authenticate(const String & user_name, const String & password) const
+    Result authenticate(const String & user_name, const String & password, const std::unordered_map<String, String> & headers) const
     {
         Poco::Net::HTTPRequest request{
             Poco::Net::HTTPRequest::HTTP_GET, this->getURI().getPathAndQuery(), Poco::Net::HTTPRequest::HTTP_1_1};
+
+        for (const auto & k : this->getHeaders())
+        {
+            auto it = headers.find(k);
+            if (it == headers.end())
+            {
+                continue;
+            }
+            request.add(k, it->second);
```

Comment:
nit, but seems unnecessary `continue` statement when it's even more readable without it, I think

```suggestion
            if (it != headers.end())
                request.add(k, it->second);
```